test/GenericMerge.test.ts

```diff
@@ -0,0 +1,102 @@
+import { expect } from "chai";
+import { ethers } from "hardhat";
+
+import { DETERMINISTIC_MIN_HEIGHT } from "./constants";
+
+describe("Generic merge contract", () => {
```

Comment:
I think BroEater tests are missing, we have geneticEater tests but I see that it is not being used so far.